TCAT/Controllers/RouteDetailDrawerViewController+Extensions.swift

```diff
@@ -101,43 +108,59 @@ extension RouteDetailDrawerViewController: UITableViewDataSource {
-                cell.configure(
-                    for: direction,
-                    isFirstStep: indexPath.row == 0,
-                    isExpanded: expandedDirections.contains(direction),
-                    delegate: self
-                )
+        
+        switch sections[indexPath.section] {
+        case .routeDetail:
+            switch directionsAndVisibleStops[indexPath.row] {
+            case .busStop(let busStop):
+                guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.busStopDetailCellIdentifier) as? BusStopTableViewCell
+                    else { return UITableViewCell()}
+                cell.configure(for: busStop.name)
                 return format(cell)
+            case .direction(let direction):
+                if direction.type == .walk || direction.type == .arrive {
+                    guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.smallDetailCellIdentifier, for: indexPath) as? SmallDetailTableViewCell
+                        else { return UITableViewCell() }
+                    cell.configure(for: direction,
+                                   isFirstStep: indexPath.row == 0,
+                                   isLastStep: indexPath.row == directionsAndVisibleStops.count - 1)
+                    return format(cell)
+                } else {
+                    let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.largeDetailCellIdentifier) as! LargeDetailTableViewCell
+                    cell.configure(
+                        for: direction,
+                        isFirstStep: indexPath.row == 0,
+                        isExpanded: expandedDirections.contains(direction),
+                        delegate: self
+                    )
+                    return format(cell)
+                }
             }
+        case .notification:
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.Cells.notificationToggleCellIdentifier) as? NotificationToggleTableViewCell
+                else { return UITableViewCell() }
+            let notifTitle = notificationTitles[indexPath.row]
+            if indexPath.row == 0 {
+                cell.setupFirstCellHairline()
+            }
+            cell.configure(for: notifTitle)
+            return format(cell)
         }
     }
 }
 
 extension RouteDetailDrawerViewController: UITableViewDelegate {
 
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        if let direction = directionsAndVisibleStops[indexPath.row].getDirection(),
-                direction.type == .depart || direction.type == .transfer {
-            return UITableView.automaticDimension
-        } else {
-            return RouteDetailCellSize.smallHeight
+        switch sections[indexPath.section] {
+        case .routeDetail:
+            if let direction = directionsAndVisibleStops[indexPath.row].getDirection(),
```

Comment:
I feel kinda kinda uneasy about this `getDirection` function. Ima look into it a little more. This is not a blocking comment just wanted to put that out there